extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/MTLS.java

```diff
@@ -0,0 +1,254 @@
+        return request(null);
+    }
+
+    /**
+     * Creates the mutual TLS client authentication mechanism.
+     * This mechanism always require the client authentication.
+     *
+     * @param tlsConfigurationName see {@link Builder#tls(String)}
+     * @return MtlsAuthenticationMechanism
+     * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information
+     */
+    static MtlsAuthenticationMechanism required(String tlsConfigurationName) {
+        return builder().tls(tlsConfigurationName).build();
+    }
+
+    /**
+     * Creates the mutual TLS client authentication mechanism.
+     * This mechanism always require the client authentication.
+     *
+     * @return MtlsAuthenticationMechanism
+     * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information
+     */
+    static MtlsAuthenticationMechanism required() {
+        return required(null);
+    }
+
+    /**
+     * @return Builder for the mutual TLS client authentication mechanism.
+     *         By default, this builder creates a mechanism which requires client authentication.
+     */
+    static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * The mutual TLS client authentication mechanism builder.
+     */
+    final class Builder {
+
+        private ClientAuth clientAuth;
+        private Function<X509Certificate, Set<String>> certificateToRolesMapper;
+        private String certificateAttribute;
+        private Map<String, Set<String>> certificateAttributeValueToRoles;
+        private Optional<String> httpServerTlsConfigName;
+
+        public Builder() {
+            this.certificateAttribute = null;
+            this.certificateToRolesMapper = null;
+            this.clientAuth = ClientAuth.REQUIRED;
+            this.certificateAttributeValueToRoles = null;
+            this.httpServerTlsConfigName = Optional.empty();
+        }
+
+        /**
+         * Configures the name of the TLS configuration used by the HTTP server for the TLS communication.
+         * Please note that this method is mutually exclusive with the 'quarkus.http.tls-configuration-name'
+         * configuration property.
+         *
+         * @param tlsConfigurationName The name of the TLS configuration to use.
+         * @return Builder
+         * @see VertxHttpConfig#tlsConfigurationName() for more information
+         */
+        public Builder tls(String tlsConfigurationName) {
+            this.httpServerTlsConfigName = Optional.ofNullable(tlsConfigurationName);
+            return this;
+        }
+
+        /**
+         * The client authentication is accepted if presented by a client.
+         * Use this method if the client authentication is only required for certain routes
+         * and secure these routes with HTTP permissions or standard security annotations.
+         *
+         * @return Builder
+         * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information
+         */
+        public Builder request() {
```

Comment:
@michalvavrik It might be confusing for some users that only `request()` is present, while indeed, I agree that having both `request()` and `required()` would not be great in the builder. 

I propose to have a method like `authentication(ClientAuth)` and doc that calling this method is not needed if the default request mode is required...